### PR TITLE
Subject prefix

### PR DESCRIPTION
**Pull request by LaVibeX**
### Description

- Email Subject Prefix is Hardcoded rightnow `.subject("[Dependency-Track] " + notification.getTitle())`, 
- This is a solution to allow the User to modify the Subject Prefix to its will  `.subject(smtpPrefix + " " + notification.getTitle() )`

This modification would be available in `Administration>Configuration>Email` 
and ` "[Dependency-Track]"` would be the first default value. 

If the User change the Subject Prefix, it will need to press Update button in order to save the changes.
Subject Prefix is **not a required field** so leaving the input box in blank is also an option.

 **_This new feature helps to distinguish between several instances of Dependency-Track, e.g. prod and qual systems._**

### Default value: _[Dependency-Track]_
image
### Modified Value:  _[Dependency-Track TEST ®]_
image
### Modified Value: " "
image

This PR depends on its corresponding PR in the **frontend** repository
https://github.com/DependencyTrack/frontend/pull/720

### Addressed Issue
This fixes https://github.com/DependencyTrack/dependency-track/issues/1823

### Additional Details 
Check if smtpPrefix is null in order to clean the value to " "
`smtpPrefix = smtpPrefix == null ? " " : smtpPrefix;`

This PR is supervised by @lukas-braune.

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

Signed-off-by: Andres Tito <andres.tito@rohde-schwarz.com>

### CODE REVIEW

File: src/main/java/org/dependencytrack/notification/publisher/SendMailPublisher.java [hunk 1]
```diff
@@ -136,7 +140,7 @@ private void sendNotification(final PublishContext ctx, Notification notificatio
             final SendMail sendMail = new SendMail()
                     .from(smtpFrom)
                     .to(destinations)
-                    .subject("[Dependency-Track] " + notification.getTitle())
+                    .subject(smtpPrefix + " " + notification.getTitle() )
```
Comment (valentijnscholten) [inline src/main/java/org/dependencytrack/notification/publisher/SendMailPublisher.java:143]: Might be nitpicking, but I suggest to change the name of the variable and setting to something more generic like `emailSubjectPrefix` or something like that. The prefix is not really tied to the SMTP protocol. If other email delivery methods are added in the future, the prefix might still be valid to use, but should not be called `smtpPrefix`.

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

**Review by nscuro — APPROVED** [review]
Thanks @LaVibeX!

**Comment by nscuro** [comment]
Looks like this broke `SendMailPublisherTest`. Do you mind having a look @LaVibeX?

**Review by nscuro — APPROVED** [review]

**Comment by girtsn** [comment]
it says that it implements https://github.com/DependencyTrack/dependency-track/issues/1823
from the ticket "add the notification subject configuration to the notification template and use same params as there exists for content"
is it possible to use variables now as in body of the notification (e.g. severity / others)? the description of this ticket seems to be about adding a static prefix only?

**Comment by nscuro** [comment]
You are right @girtsn, I reopened the issue.

**Pull request merged**